scipy/stats/_qmc.py

```diff
@@ -87,6 +87,22 @@ def check_random_state(seed=None):
                          ' numpy.random.Generator instance')
 
 
+def _bounds_iv(l_bounds, u_bounds, d):
+    """Bounds input validation."""
+    lower = np.atleast_1d(l_bounds)
+    upper = np.atleast_1d(u_bounds)
+
+    lower, upper = np.broadcast_arrays(lower, upper)
+
+    if len(lower) != d:
+        raise ValueError('Sample dimension is different than bounds dimension')
```

Comment:
Ultimately, don't `lower` and `upper` just need to be broadcastable to the shape of `d`? Maybe
```suggestion
    lower = np.broadcast_to(lower, d)
    upper = np.broadcast_to(upper, d)
```
If they aren't broadcastable, you get a reasonable message, but you could also `try`-`except` for a more specific one.

Reply:
Yes I added a `try-except` and modified the message accordingly.